core/src/main/java/org/frankframework/core/Adapter.java

```diff
@@ -602,6 +761,7 @@ public PipeLineResult processMessageDirect(String messageId, Message message, Pi
 			} catch (Throwable t) {
 				log.warn("Adapter [{}] error processing message with ID [{}]", name, messageId, t);
 				result.setState(ExitState.ERROR);
+				result.setExitCode(pipeLineSession.get(PipeLineSession.EXIT_CODE_CONTEXT_KEY, 500)); // If there was an exception that was not handled by the pipeline, consider it an internal server error.
```

Comment:
Dit maakt het mogelijk dat een pipeline aan de caller een exit-code kan geven, en dat er bij een error een standard exit-code 500 is.
Dit is hetzelfde als bij Receiver, maar deze method wordt gebruikt wanneer de adapter wordt aangeroepen zonder receiver.